lib/AST/ArchetypeBuilder.cpp

```diff
@@ -912,6 +939,63 @@ bool ArchetypeBuilder::addConformanceRequirement(PotentialArchetype *PAT,
   return false;
 }
 
+bool ArchetypeBuilder::addLayoutRequirement(PotentialArchetype *PAT,
+                                            ProtocolDecl *Proto,
+                                            RequirementSource Source) {
+  llvm::SmallPtrSet<ProtocolDecl *, 8> Visited;
+  return addLayoutRequirement(PAT, Proto, Source, Visited);
+}
+
+bool ArchetypeBuilder::addLayoutRequirement(PotentialArchetype *PAT,
+                                            ProtocolDecl *Proto,
+                                            RequirementSource Source,
+                               llvm::SmallPtrSetImpl<ProtocolDecl *> &Visited) {
+  // Add the requirement to the representative.
```

Comment:
This is just copy and pasted from addConformanceRequirement(). I don't think layouts should be protocols